common/device-model.cpp

```diff
@@ -2812,163 +2964,231 @@ namespace rs2
                             ImGui::PopFont();
                             throw;
                         }
-                    });
+                        });
 
-                    label = rsutils::string::from() << "Post-Processing##" << id;
+                    label = rsutils::string::from() << pb->get_name() << "##" << id;
                     if (ImGui::TreeNode(label.c_str()))
                     {
-                        for (auto&& pb : sub->post_processing)
+                        ImGui::SetCursorPosY(ImGui::GetCursorPosY() + 5);
+                        pb->draw_options(viewer,
+                            dev.is< playback >() || update_read_only_options,
+                            false,
+                            error_message);
+
+                        ImGui::TreePop();
+                    }
+                }
+                ImGui::TreePop();
+            }
+        }
+    }
+
+    void device_model::draw_embedded_filters(std::shared_ptr<subdevice_model> sub, float windows_width,
```

Comment:
Very similar to post processing handling. Consider creating a common implementation, passing relevant filters and enable flag as parameters.